drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,502 @@
+	const struct nxp_irtc_config *config = dev->config;
+	struct nxp_irtc_data *data = dev->data;
+	RTC_Type *irtc_reg = config->base;
+	uint16_t curr_alarm_mask = data->alarm_mask;
+
+	if (id != 0 || !timeptr) {
+		return -EINVAL;
+	}
+
+	*mask = 0;
+	if ((curr_alarm_mask & RTC_ALARM_TIME_MASK_SECOND) == RTC_ALARM_TIME_MASK_SECOND) {
+		timeptr->tm_sec = ((irtc_reg->ALM_SECONDS & RTC_ALM_SECONDS_ALM_SEC_MASK) >>
+				   RTC_ALM_SECONDS_ALM_SEC_SHIFT);
+		*mask |= RTC_ALARM_TIME_MASK_SECOND;
+	}
+
+	if ((curr_alarm_mask & RTC_ALARM_TIME_MASK_MINUTE) == RTC_ALARM_TIME_MASK_MINUTE) {
+		timeptr->tm_min = ((irtc_reg->ALM_HOURMIN & RTC_ALM_HOURMIN_ALM_MIN_MASK) >>
+				   RTC_ALM_HOURMIN_ALM_MIN_SHIFT);
+		*mask |= RTC_ALARM_TIME_MASK_MINUTE;
+	}
+
+	if ((curr_alarm_mask & RTC_ALARM_TIME_MASK_HOUR) == RTC_ALARM_TIME_MASK_HOUR) {
+		timeptr->tm_hour = ((irtc_reg->ALM_HOURMIN & RTC_ALM_HOURMIN_ALM_HOUR_MASK) >>
+				    RTC_ALM_HOURMIN_ALM_HOUR_SHIFT);
+		*mask |= RTC_ALARM_TIME_MASK_HOUR;
+	}
+
+	if ((curr_alarm_mask & RTC_ALARM_TIME_MASK_MONTHDAY) == RTC_ALARM_TIME_MASK_MONTHDAY) {
+		timeptr->tm_mday = ((irtc_reg->ALM_DAYS & RTC_ALM_DAYS_ALM_DAY_MASK) >>
+				    RTC_ALM_DAYS_ALM_DAY_SHIFT);
+		*mask |= RTC_ALARM_TIME_MASK_MONTHDAY;
+	}
+
+	if ((curr_alarm_mask & RTC_ALARM_TIME_MASK_MONTH) == RTC_ALARM_TIME_MASK_MONTH) {
+		timeptr->tm_mon = ((irtc_reg->ALM_YEARMON & RTC_ALM_YEARMON_ALM_MON_MASK) >>
+				   RTC_ALM_YEARMON_ALM_MON_SHIFT) -
+				  1;
+		*mask |= RTC_ALARM_TIME_MASK_MONTH;
+	}
+
+	if ((curr_alarm_mask & RTC_ALARM_TIME_MASK_YEAR) == RTC_ALARM_TIME_MASK_YEAR) {
+		timeptr->tm_year =
+			(int8_t)((irtc_reg->ALM_YEARMON & RTC_ALM_YEARMON_ALM_YEAR_MASK) >>
+				 RTC_ALM_YEARMON_ALM_YEAR_SHIFT) +
+			RTC_NXP_IRTC_YEAR_OFFSET;
+		*mask |= RTC_ALARM_TIME_MASK_YEAR;
+	}
+
+	return 0;
+}
+
+static int nxp_irtc_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+	struct nxp_irtc_data *data = dev->data;
+	int ret = data->is_alarm_pending;
+
+	if (id != 0) {
+		return -EINVAL;
+	}
+
+	data->is_alarm_pending = 0;
+
+	return ret;
+}
+
+static int nxp_irtc_alarm_set_callback(const struct device *dev, uint16_t id,
+				       rtc_alarm_callback callback, void *user_data)
+{
+	struct nxp_irtc_data *data = dev->data;
+
+	if (id != 0) {
+		return -EINVAL;
+	}
+
+	data->alarm_callback = callback;
+	data->alarm_user_data = user_data;
+
+	return 0;
+}
+
+#endif /* CONFIG_RTC_ALARM */
+#if defined(CONFIG_RTC_UPDATE) || defined(__DOXYGEN__)
+
+static int nxp_irtc_update_set_callback(const struct device *dev, rtc_update_callback callback,
+					void *user_data)
+{
+	struct nxp_irtc_data *data = dev->data;
+
+	data->update_callback = callback;
+	data->update_user_data = user_data;
+
+	return 0;
+}
+
+#endif /* CONFIG_RTC_UPDATE */
+#if defined(CONFIG_RTC_CALIBRATION) || defined(__DOXYGEN__)
+
+static int nxp_irtc_set_calibration(const struct device *dev, int32_t calibration)
+{
+	struct nxp_irtc_data *data = dev->data;
+
+	data->calibration = calibration;
```

Comment:
agree with bjarki to make this return enotsup, this doesnt look like it does anything at all, let's not give the illusion of a calibration feature being implemented (why is it not implemented?)